source/common/buffer/buffer_impl.cc

```diff
@@ -147,6 +147,15 @@ void OwnedImpl::commit(RawSlice* iovecs, uint64_t num_iovecs) {
       }
     }
 
+    // If the slices just committed are not at the very end of the buffer,
```

Comment:
@brian-pane IIRC the libevent implementation doesn't even require calling commit(). It will just return the previous result of reserve() effectively. WDYT about doing something similar here so that we don't even explicitly keep track of reservation state? Seems simpler?

Reply:
This PR supports that case too: https://github.com/envoyproxy/envoy/pull/6663/commits/7ba70bbd20bbb0c10c8ec6002a69594ec1e30855#diff-7ac35a12fc96d0b129f9541cfa0eafbcR421